internal/cloud/realm/client.go

```diff
@@ -91,5 +91,56 @@ func (c *client) do(method, path string, options api.RequestOptions) (*http.Resp
 		req.Header.Set(api.HeaderAuthorization, "Bearer "+auth)
 	}
 	client := &http.Client{}
-	return client.Do(req)
+	res, err := client.Do(req)
+	if err != nil {
+		return res, err
+	}
+	if res.StatusCode != http.StatusUnauthorized || !options.UseAuth || options.PreventRefresh {
```

Comment:
let's change this check to just see if `res.StatusCode` is in between 200 and 299, and if it is, then we can just return `res, nil` here